pep-0716.rst

```diff
@@ -0,0 +1,319 @@
+
+
+Backwards Compatibility
+=======================
+
+This PEP breaks compatibility in a few ways:
+
+* Tools that are currently emitting filenames where ``name`` has been normalized
+  in accordance with the current spec are immediately no longer compliant and
+  must be updated to emit non-normalized names.
+
+  * This is mitigated by the fact that all tools are required to continue to
+    accept both normalized and non-normalized filenames unless they *know* that
+    they only work on *new* distributions (PyPI uploads, ``pyproject-build``, etc).
+
+* Tools that emit normalized names into ``METADATA``, ``PKG-INFO``, or when
+  uploading to a repository are immediately no longer compliant and must be
+  updated to emit non-normalized names.
+
+  * It's unclear in the current spec whether names were intended to be normalized
+    in this case or not, but the practice of normalization here has caused a
+    number of people to be confused why their names are different from what
+    they've entered.
+
+* Tools that are currently emitting the names in the simple API (outside of the URL
+  itself) as normalized, which is either allowed or required by the spec
+  currently are immediately not longer complaint and must be updated to emit
+  non-normalized names.
+
+  * Like for filenames, this is mitigated by the fact that all tools are required
+    to continue to accept both normalized and non-normalized values.
+
+
+Tools that validate *new* values should ideally start warning on now-invalid
+options for some period of time, before starting to hard fail when encountering
+them.
+
+
+Rejected Ideas
+==============
+
+Require Normalization Everywhere
+--------------------------------
+
+One other possible idea is to simply require normalization everywhere, however
+this PEP rejects that.
+
+The primary reason we reject it is that the name of a project is not an internal
+identifier, but is central to that project's identity. Projects often have
+strong opinions on the way that their project's name should look, and
+normalization removes that from them.
+
+There are situations where we need a normalized value, so this PEP does use
+them, but attempts to use them sparingly, only when they're actually required.
+It treats normalization as something that is done when software is talking to
+software about a project, and not when humans are talking about it.
+
+
+Require Normalization in Filenames
+----------------------------------
+
+Filenames sit in a weird place, in most cases they are produced by software
+and are consumed by software, so in theory it should be fine to normalize them
+which has some nice properties.
+
+However, this PEP rejects doing that.
+
+Although they are often a software-to-software identifier, they are also used by
+humans when sharing and manually downloading the software. They appear in places
+like the PyPI UI, GitHub Releases, downstream Linux repositories, etc. In some
```

Comment:
Downstream Linux repositories? Huh? Linux distros as a rule don't expose to users files straight off PyPI, they repack them into their own archive format following their own (nearly always normalized) naming convention. Furthermore, AFAIK most if not nearly all have naming and normalization policies similar or substantially stricter than PEP 503's for the canonical user-facing name of their packages, not just the file names. Therefore, I'm rather confused how this is relevant here.

Reply:
Also, the wheel filename was *explicitly designed* to be for machine consumption, so saying it's not a software-to-software identifier is misrepresenting the intention of the design. (Thanks to compatibility tags, no-one would say wheel filenames are for human consumption 😉)

IMO, this whole argument is the weakest point of the whole PEP, and the overall proposal would be changed very little if it was changed to require normalisation of the project name in wheel and sdist filenames, but was left otherwise unchanged. If that isn't actually the case, then this is the section that should explain what the *actual* problem with normalising is. That would be a far better rejection reason than the current (frankly, largely subjective) argument.

Reply:
Yup, entirely agreed—I'd briefly mentioned it in my top-level review, but particularly as a PEP editor decided to save my own detailed commentary on this for the PEP discussion thread. However, as it ended up getting discussed at some length on Barry's thread that kicked off this PEP, it sounds like Donald mentioned there that he'd be revising this accordingly (hopefully to require normalization, which seemed to be the general consensus, rather than just allow it).